src/mongocxx/collection.cpp

```diff
@@ -426,12 +426,89 @@ std::int64_t collection::count(bsoncxx::document::view filter, const options::co
 }
 
 bsoncxx::document::value collection::create_index(bsoncxx::document::view keys,
-                                                  bsoncxx::document::view options) {
+                                                  const options::index& options) {
     scoped_bson_t bson_keys{keys};
+    scoped_bson_t weights{options.weights()};
+    scoped_bson_t partial_filter_expression{options.partial_filter_expression()};
     bson_error_t error;
+    mongoc_index_opt_geo_t geo_opt{};
+    mongoc_index_opt_t opt{};
+    mongoc_index_opt_wt_t wt_opt{};
+    libmongoc::index_opt_init(&opt);
+
+    if (options.background()) {
+        opt.background = options.background().value();
+    }
+    if (options.unique()) {
+        opt.unique = options.unique().value();
+    }
+    if (options.name()) {
+        opt.name = options.name().value().c_str();
+    }
+    if (options.sparse()) {
+        opt.sparse = options.sparse().value();
+    }
+    if (options.storage_options()) {
+        const options::index::wiredtiger_storage_options* wt_options;
+        libmongoc::index_opt_wt_init(&wt_opt);
+        switch (options.storage_options()->type()) {
+            case 1:
+                wt_options = dynamic_cast<const options::index::wiredtiger_storage_options*>(options.storage_options());
```

Comment:
if you are already sure that it is a wiredtiger_storage_opt_t, you can just reinterpret_cast